### ISSUE TITLE
Implementation of JEI's IRecipeCategory results in error with Just Enough Calculation

### ISSUE THREAD

**Issue by whizzball1** [Low-Drag-MC/LDLib-MultiLoader#23]
Just Enough Calculation, a mod like EMI but with an 'inventory' that can accept recipes from JEI, encounters serious errors attempting to transfer GregTech recipes from JEI. It cannot do so because it relies on all JEI recipes somehow going back to the Minecraft `Recipe` type. When GregTech registers its JEI `RecipeTypeCategory`s [here](https://github.com/GregTechCEu/GregTech-Modern/blob/1.20.1/src/main/java/com/gregtechceu/gtceu/integration/jei/recipe/GTRecipeTypeCategory.java), it uses LDLib's [`ModularUIRecipeCategory`](https://github.com/Low-Drag-MC/LDLib-Architectury/blob/1.20.1/common/src/main/java/com/lowdragmc/lowdraglib/jei/ModularUIRecipeCategory.java).

`ModularUIRecipeCategory` accepts a T type `T extends ModularWrapper<?>` which, notably, does not in any way extend Minecraft's `Recipe` type.

In contrast, other mods like EnderIO implement `IRecipeCategory` using a type that extends `Recipe`. An example is [`TankCategeory`](https://github.com/Team-EnderIO/EnderIO/blob/dev/1.20.4/src/machines/java/com/enderio/machines/common/integrations/jei/category/TankCategory.java), which accepts [`TankRecipe`](https://github.com/Team-EnderIO/EnderIO/blob/dev/1.20.4/src/machines/java/com/enderio/machines/common/recipe/TankRecipe.java) which extends `Recipe` (through [`EnderRecipe`](https://github.com/Team-EnderIO/EnderIO/blob/dev/1.20.4/src/core/java/com/enderio/core/common/recipes/EnderRecipe.java)).

It happens that GregTech's actual recipes ([`GTRecipe`](https://github.com/GregTechCEu/GregTech-Modern/blob/1.20.1/src/main/java/com/gregtechceu/gtceu/api/recipe/GTRecipe.java)) do extend `Recipe`, but in order to use LDLib's JEI Recipe Category feature, it sticks its perfectly normal `Recipe`s into the class [`GTRecipeWrapper`](https://github.com/GregTechCEu/GregTech-Modern/blob/1.20.1/src/main/java/com/gregtechceu/gtceu/integration/jei/recipe/GTRecipeWrapper.java) which extends LDLib's `ModularWrapper<Widget>`.

TL;DR: LDLib implements JEI's `IRecipeCategory` in a way that does not ultimately reference Minecraft's `Recipe` type. This causes problems for any mods which want to work universally with all recipes, like Just Enough Calculation.

**Comment by whizzball1** [comment]
(Note: I just mentioned this issue in a GregTech issue, shown above, so they know what's happening.)
